lib/src/facts/aix/memory_resolver.cc

```diff
@@ -0,0 +1,47 @@
+#include <internal/facts/aix/memory_resolver.hpp>
+#include <internal/util/aix/odm.hpp>
+
+#include <sys/vminfo.h>
+#include <sys/cfgodm.h>
+#include <system_error>
+
+using namespace std;
+using namespace facter::util::aix;
+
+namespace facter { namespace facts { namespace aix {
+    memory_resolver::data memory_resolver::collect_data(collection& facts)
+    {
+        data result;
+
+        vminfo info;
+        auto res = vmgetinfo(&info, VMINFO, sizeof(info));
+        if (res < 0) {
+            throw system_error(errno, system_category());
+        }
+        result.mem_total = info.memsizepgs * 4096;
+        result.mem_free = info.numfrb * 4096;
+
+        auto cu_at_query = odm_class<CuAt>::open("CuAt").query("value=paging and attribute=type");
```

Comment:
Yeah. the odm_class type just wraps an existing ODM struct. cfgodm provides the ODM structs used in the AIX system configuration, including CuAt.